src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -3420,4 +3424,264 @@ class C : I
+
+        var comp = CreateCompilation(
+            new[] { (source, path), s_attributesSource },
+            parseOptions: RegularWithInterceptors,
+            options: TestOptions.DebugExe.WithSourceReferenceResolver(
+                new SourceFileResolver(ImmutableArray<string>.Empty, null, pathMap)));
+        comp.VerifyEmitDiagnostics(PlatformInformation.IsWindows
+                // C:\My\Machine\Specific\Path\Program.cs(11,25): error CS27008: Cannot intercept: Path 'C:\My\Machine\Specific\Path\Program.cs' is unmapped. Expected mapped path '/_/Program.cs'.
+                //     [InterceptsLocation(@"C:\My\Machine\Specific\Path\Program.cs", 5, 3)]
+                ? Diagnostic(ErrorCode.ERR_InterceptorPathNotInCompilationWithUnmappedCandidate, @"@""C:\My\Machine\Specific\Path\Program.cs""").WithArguments(@"C:\My\Machine\Specific\Path\Program.cs", "/_/Program.cs").WithLocation(11, 25)
+
+                // /My/Machine/Specific/Path/Program.cs(11,25): error CS27008: Cannot intercept: Path '/My/Machine/Specific/Path/Program.cs' is unmapped. Expected mapped path '/_/Program.cs'.
+                //     [InterceptsLocation(@"/My/Machine/Specific/Path/Program.cs", 5, 3)]
+                : Diagnostic(ErrorCode.ERR_InterceptorPathNotInCompilationWithUnmappedCandidate, @"@""/My/Machine/Specific/Path/Program.cs""").WithArguments("/My/Machine/Specific/Path/Program.cs", "/_/Program.cs").WithLocation(11, 25)
+            );
+    }
+
+    [Fact]
+    public void PathMapping_03()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            C c = new C();
+            c.M();
+
+            class C
+            {
+                public void M() => throw null!;
+
+                [InterceptsLocation(@"\_\Program.cs", 5, 3)]
+                public void Interceptor() => Console.Write(1);
+            }
+            """;
+        var pathPrefix = PlatformInformation.IsWindows ? """C:\My\Machine\Specific\Path\""" : "/My/Machine/Specific/Path/";
+        var path = pathPrefix + "Program.cs";
+        var pathMap = ImmutableArray.Create(new KeyValuePair<string, string>(pathPrefix, "/_/"));
+
+        var comp = CreateCompilation(
+            new[] { (source, path), s_attributesSource },
+            parseOptions: RegularWithInterceptors,
+            options: TestOptions.DebugExe.WithSourceReferenceResolver(
+                new SourceFileResolver(ImmutableArray<string>.Empty, null, pathMap)));
+        comp.VerifyEmitDiagnostics(
+            // C:\My\Machine\Specific\Path\Program.cs(11,25): error CS27002: Cannot intercept: compilation does not contain a file with path '\_\Program.cs'.
+            //     [InterceptsLocation(@"\_\Program.cs", 5, 3)]
+            Diagnostic(ErrorCode.ERR_InterceptorPathNotInCompilation, @"@""\_\Program.cs""").WithArguments(@"\_\Program.cs").WithLocation(11, 25));
```

Comment:
Ideally we could try doing another error recovery search by normalizing slashes in the path we have, then searching for it. We could then give a more specific diagnostic here.